File: Npc/11001076.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11001076: Slacky
/// </summary>
public class _11001076 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0831180407003671$
    // - Have you been to $map:02000051$?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0831180407003674$
                // - One day, I'm going to become as great a weaponsmith as my big brother!
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11001076.py
""" 11001076: Slacky """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407003671$
        # - Have you been to $map:02000051$?
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:0831180407003674$
        # - One day, I'm going to become as great a weaponsmith as my big brother!
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.CLOSE
        return Option.NONE
